### PR TITLE
Chore: Add errutils helpers

### PR DESCRIPTION
**Pull request by marefr**
**What is this?**

Add helpers for the errutil package in favor of `errutil.NewBase`. Reads cleaner I think and reduces length a bit defining these. We could potentially/eventually make `errutil.NewBase` private given we cover all available statuses with helper functions.

Before:
`errutil.NewBase(errutil.StatusBadRequest, "<msg id>", <opts...>)`

After:
`errutil.BadRequest("<msg id>", <opts...>)`

Yay or nay?

Came up when starting looking into #68480

### CODE REVIEW

File: pkg/util/errutil/errors.go [hunk 1]
```diff
@@ -41,6 +41,76 @@ func NewBase(reason StatusReason, msgID string, opts ...BaseOpt) Base {
...
+}
+
+// Unknown initializes a new [Base] error with reason StatusUnknown.
+//
+// See NewBase for more information.
+func Unknown(msgID string, opts ...BaseOpt) Base {
+	return NewBase(StatusUnknown, msgID, opts...)
+}
+
+// Unknown initializes a new [Base] error with reason StatusUnknown.
```
Comment (mildwonkey) [inline pkg/util/errutil/errors.go:79]:
```suggestion
// Unknown initializes a new [Base] error with reason StatusTimeout.
```

**Review by mildwonkey — APPROVED** [review]
I think this is a great enhancement! I left the tiniest of nitpicks but it's not a blocker (just a comment)

File: pkg/util/errutil/errors.go (hunk 1 above)
Comment (marefr) [reply pkg/util/errutil/errors.go:79]: Nice spotted. Managed to miss the function name as well :) 

```suggestion
// Timeout initializes a new [Base] error with reason StatusTimeout.
```

**Comment by ryantxu** [comment]
nice!  any thoughts on moving errorutils somewhere we can use it from plugins?

**Review by Jguer — APPROVED** [review]
:+1:

**Review by zserge — APPROVED** [review]

**Comment by marefr** [comment]
> nice! any thoughts on moving errorutils somewhere we can use it from plugins?

@ryantxu no, not currently. Thinking app platform plugins or making use of the statuses for current app/datasource backend plugins?

**Review by wbrowne — APPROVED** [review]
🚀

**Pull request merged**